projects/core/src/auth/http-interceptors/token-revocation.interceptor.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { EMPTY, Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import {
+  InterceptorUtil,
+  TOKEN_REVOCATION,
+} from '../../occ/utils/interceptor-util';
+
+@Injectable({ providedIn: 'root' })
+export class TokenRevocationInterceptor implements HttpInterceptor {
+  constructor() {}
+
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    const isInterceptingTokenRevocationRequest = this.isTokenRevocationRequest(
+      request
+    );
+    if (isInterceptingTokenRevocationRequest) {
+      request = InterceptorUtil.removeHeader(TOKEN_REVOCATION, request);
+    }
+
+    return next.handle(request).pipe(
+      catchError((response: any) => {
```

Comment:
LGTM

code wise, I'd do, but I guess it's a matter of taste ;)
```typescript
const hasRevokeHeader = InterceptorUtil.getInterceptorParam<string>(TOKEN_REVOCATION, request.headers);

if (response instanceof HttpErrorResponse && !!hasRevokeHeader) {
...
}
```

Reply:
In the end, I left the function because I thought improved the testability and could also potentially serve as an extension hook.  Although I did simplify the error handling like so: 
```
      catchError((error: any) => {
        if (isTokenRevocationRequest) {
          return EMPTY;
        }
        return throwError(error);
      })
```